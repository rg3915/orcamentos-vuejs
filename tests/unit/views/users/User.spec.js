import { mount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import CoreuiVue from '@coreui/vue'
import User from '@/views/users/User'
import appRouter from '@/router'

const localVue = createLocalVue()
localVue.use(VueRouter)
const router = appRouter
router.push({path: '/users/1'})

localVue.use(CoreuiVue)

describe('User.vue', () => {
  let wrapper
  beforeEach(() => {
    wrapper = mount(User, {
      localVue,
      router
    })
  })
  it('has a name', () => {
    expect(User.name).toBe('User')
  })
  it('is Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })
  it('is User', () => {
    expect(wrapper.findComponent(User)).toBeTruthy()
  })
  it('should have methods', () => {
    expect(typeof User.methods.goBack).toEqual('function')
  })
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
