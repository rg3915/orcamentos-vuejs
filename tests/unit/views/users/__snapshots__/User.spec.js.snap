// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User.vue renders correctly 1`] = `
<div
  class="row"
>
  <div
    class="col-12 col-lg-6"
  >
    <div
      class="card"
    >
      <header
        class="card-header"
      >
        
        User id:  1
      
      </header>
       
      <div
        class="card-body"
      >
        <div
          data-v-1c2db5c2=""
          small=""
        >
          <!---->
          <div
            class="position-relative table-responsive"
            data-v-1c2db5c2=""
          >
            <table
              class="table table-striped table-fixed"
              data-v-1c2db5c2=""
            >
              <thead
                data-v-1c2db5c2=""
              >
                <tr
                  data-v-1c2db5c2=""
                >
                  <th
                    class=""
                    data-v-1c2db5c2=""
                    style="vertical-align: middle; overflow: hidden; width: 150px;"
                  >
                    <div
                      data-v-1c2db5c2=""
                    >
                      Samppa Nori
                    </div>
                    <!---->
                  </th>
                  <th
                    class=""
                    data-v-1c2db5c2=""
                    style="vertical-align: middle; overflow: hidden; width: 150px;"
                  >
                    <div
                      data-v-1c2db5c2=""
                    >
                      
                    </div>
                    <!---->
                  </th>
                </tr>
                <!---->
              </thead>
              <tbody
                class="position-relative"
                data-v-1c2db5c2=""
              >
                <tr
                  data-v-1c2db5c2=""
                >
                  <td
                    class=""
                    data-v-1c2db5c2=""
                  >
                     registered 
                  </td>
                  <td
                    class=""
                    data-v-1c2db5c2=""
                  >
                     2012/01/01 
                  </td>
                </tr>
                <!---->
                <tr
                  data-v-1c2db5c2=""
                >
                  <td
                    class=""
                    data-v-1c2db5c2=""
                  >
                     role 
                  </td>
                  <td
                    class=""
                    data-v-1c2db5c2=""
                  >
                     Member 
                  </td>
                </tr>
                <!---->
                <tr
                  data-v-1c2db5c2=""
                >
                  <td
                    class=""
                    data-v-1c2db5c2=""
                  >
                     status 
                  </td>
                  <td
                    class=""
                    data-v-1c2db5c2=""
                  >
                     Active 
                  </td>
                </tr>
                <!---->
                <!---->
              </tbody>
              <!---->
            </table>
            <!---->
          </div>
          <!---->
        </div>
      </div>
       
      <footer
        class="card-footer"
      >
        <button
          class="btn btn-primary"
          type="button"
        >
          Back
        </button>
      </footer>
    </div>
  </div>
</div>
`;
